src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActor.scala

```diff
@@ -97,13 +101,19 @@ private[impl] class InstanceTrackerActor(
   override def receive: Receive = initializing
 
   private[this] def initializing: Receive = LoggingReceive.withLabel("initializing") {
-    case appTasks: InstanceTracker.InstancesBySpec =>
+    case initialInstances: InstanceTracker.InstancesBySpec =>
       logger.info("Task loading complete.")
 
+      instancesBySpec = initialInstances
+      counts = TaskCounts(initialInstances.allInstances, healthStatuses = Map.empty)
```

Comment:
I guess we could make the task counts part if the `InstencesBySpec` and the also update the metrics there. What do you think?

Reply:
`the InstencesBySpec and the` — what comes after the second `the`?

Reply:
Ah, it should be `and then also update the metrics there`.

Reply:
Yes. However, if these classes are removed that would not be an option.